kube-core/src/schema.rs

```diff
@@ -111,10 +66,97 @@ impl Visitor for StructuralSchemaRewriter {
     }
 }
 
+/// Bring all property definitions from subschemas up to the root schema,
+/// since Kubernetes doesn't allow subschemas to define properties.
+fn hoist_subschema_properties(
+    subschemas: &mut Vec<Schema>,
+    common_obj: &mut Option<Box<ObjectValidation>>,
+    instance_type: &mut Option<SingleOrVec<InstanceType>>,
+    allow_if_equal: bool,
```

Comment:
I think we can get rid of `allow_if_equal` now and just assume that across the board?

Reply:
That changes the current implementation for oneOf but sounds reasonable. Adopted!